kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/BoundsTest.kt

```diff
@@ -10,24 +10,225 @@ class BoundsTest : WordSpec() {
-            }.shouldHaveMessage("Collection should have upper bound 2, but the following elements are above it: [3]")
+         "fail for IntArray" {
+            shouldThrowAny { intArrayOf(1, 2, 3) shouldHaveUpperBound 2 }
+               .shouldHaveMessage(msg("IntArray"))
+         }
+
+         "fail for LongArray" {
+            shouldThrowAny { longArrayOf(1, 2, 3) shouldHaveUpperBound 2 }
+               .shouldHaveMessage(msg("LongArray", "2", "3L"))
+         }
+
+         "fail for FloatArray" {
+            shouldThrowAny { floatArrayOf(0.0f, 0.2f, 0.3f) shouldHaveUpperBound 0.2f }
+               .shouldHaveMessage(msg("FloatArray", "0.2", "0.3f"))
+         }
+
+         "fail for DoubleArray" {
+            shouldThrowAny { doubleArrayOf(0.0, 0.2, 0.3) shouldHaveUpperBound 0.2 }
+               .shouldHaveMessage(msg("DoubleArray", "0.2", "0.3"))
+         }
+
+         "fail for Array" {
+            shouldThrowAny { arrayOf(1, 2, 3) shouldHaveUpperBound 2 }
+               .shouldHaveMessage(msg("Array"))
+         }
+
+         "fail for List" {
+            shouldThrowAny { listOf(1, 2, 3) shouldHaveUpperBound 2 }
+               .shouldHaveMessage(msg("List"))
+         }
+
+         "fail for Set" {
+            shouldThrowAny { setOf(1, 2, 3) shouldHaveUpperBound 2 }
+               .shouldHaveMessage(msg("Set"))
+         }
+
+         "fail for Range" {
+            shouldThrowAny { 1..3 shouldHaveUpperBound 2 }
+               .shouldHaveMessage(msg("Range"))
          }
       }
 
       "haveLowerBound" should {
          "pass" {
+            byteArrayOf() shouldHaveLowerBound 3
+            byteArrayOf(3) shouldHaveLowerBound 3
+            byteArrayOf(1, 2, 3) shouldHaveLowerBound 1
+            byteArrayOf(1, 2, 3) shouldHaveLowerBound 0
+
+            shortArrayOf() shouldHaveLowerBound 3
+            shortArrayOf(3) shouldHaveLowerBound 3
+            shortArrayOf(1, 2, 3) shouldHaveLowerBound 1
+            shortArrayOf(1, 2, 3) shouldHaveLowerBound 0
+
+            charArrayOf() shouldHaveLowerBound 'c'
+            charArrayOf('c') shouldHaveLowerBound 'c'
+            charArrayOf('b', 'c', 'd') shouldHaveLowerBound 'b'
+            charArrayOf('b', 'c', 'd') shouldHaveLowerBound 'a'
+
+            intArrayOf() shouldHaveLowerBound 3
+            intArrayOf(3) shouldHaveLowerBound 3
+            intArrayOf(1, 2, 3) shouldHaveLowerBound 1
+            intArrayOf(1, 2, 3) shouldHaveLowerBound 0
+
+            longArrayOf() shouldHaveLowerBound 3
+            longArrayOf(3) shouldHaveLowerBound 3
+            longArrayOf(1, 2, 3) shouldHaveLowerBound 1
+            longArrayOf(1, 2, 3) shouldHaveLowerBound 0
+
+            floatArrayOf() shouldHaveLowerBound 0.3f
+            floatArrayOf(0.3f) shouldHaveLowerBound 0.3f
+            floatArrayOf(0.1f, 0.2f, 0.3f) shouldHaveLowerBound 0.1f
+            floatArrayOf(0.1f, 0.2f, 0.3f) shouldHaveLowerBound 0f
+
+            doubleArrayOf() shouldHaveLowerBound 0.3
```

Comment:
As an example for the aforementioned guard-clauses, an empty double array will not have 0.3 as lowerbound

Reply:
Yep, I'll close this comment as resolved to keep that discussion in one thread - https://github.com/kotest/kotest/pull/4378#discussion_r1779820867